internal/resources/aws/sns_topic.go

```diff
@@ -222,3 +222,158 @@ func (r *SNSTopic) notificationsCostComponent(name, unit string, multiplier int6
+	}
+
+	return &schema.CostComponent{
+		Name:            "FIFO Publish API payload",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("AmazonSNS"),
+			ProductFamily: strPtr(""),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", Value: strPtr("F-Ingress-Tier1")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			StartUsageAmount: strPtr("0"),
+		},
+	}
+}
+func (r *SNSFIFOTopic) NotificationsCostComponent(subscriptions int64, requests *int64) *schema.CostComponent {
+	var q *decimal.Decimal
+	if subscriptions == 0 {
+		q = &decimal.Zero
+	} else if requests != nil {
+		q = decimalPtr(decimal.NewFromInt(subscriptions * *requests))
+	}
+
+	return &schema.CostComponent{
+		Name:            "FIFO notifications",
+		Unit:            "1M notifications",
+		UnitMultiplier:  decimal.NewFromInt(1000000),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("AmazonSNS"),
+			ProductFamily: strPtr(""),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", Value: strPtr("F-DA-SQS")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			StartUsageAmount: strPtr("0"),
+		},
+	}
+}
+func (r *SNSFIFOTopic) NotificationPayloadCostComponent(subscriptions int64, requests *int64, requestSizeGB *float64) *schema.CostComponent {
+	var q *decimal.Decimal
+	if subscriptions == 0 {
+		q = &decimal.Zero
+	} else if requests != nil && requestSizeGB != nil {
+		q = decimalPtr(decimal.NewFromFloat(float64(subscriptions**requests) * *requestSizeGB))
+	}
+
+	return &schema.CostComponent{
+		Name:            "FIFO notification payload",
+		Unit:            "GB",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: q,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("aws"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("AmazonSNS"),
+			ProductFamily: strPtr(""),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "usagetype", Value: strPtr("F-Egress-SQS")},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			StartUsageAmount: strPtr("0"),
+		},
+	}
+}
+
+func (r *SNSFIFOTopic) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *SNSFIFOTopic) BuildResource() *schema.Resource {
+	var requestSizeGB *float64
+	if r.RequestSizeKB != nil {
+		requestSizeGB = floatPtr(*r.RequestSizeKB / 1000000.0)
+	}
+
+	components := []*schema.CostComponent{
+		r.PublishAPIRequestsCostComponent(r.MonthlyRequests),
+		r.PublishAPIPayloadCostComponent(r.MonthlyRequests, requestSizeGB),
+		r.NotificationsCostComponent(r.Subscriptions, r.MonthlyRequests),
+		r.NotificationPayloadCostComponent(r.Subscriptions, r.MonthlyRequests, requestSizeGB),
```

Comment:
No need to pass into struct properties into a struct method. Let's remove and just use the struct properties directly in the method.